The Project

tomslw opened the pull request:
Added its abilities to get an ip with DHCP
Added the WOL packet sender function

tomslw committed:
Everything for the Ethernet adapter for now.

Added its abilities to get an ip with DHCP
Added the WOL packet sender function

Astro-Johnny commented on main/main.ino:
```diff
@@ -1,9 +1,54 @@
+      WOL_packet[o] = PC_mac[i];
+      o++;
+    }
+  }
```
Reinitialising the `WOL_packet[]` on each invocation is not really necessary. Consider setting it in `setup()` or even defining it as constant.

Astro-Johnny commented on main/main.ino:
```diff
@@ -1,9 +1,54 @@
+  }
+
+  Udp.beginPacket( brodcast_address, 9 );                       // puts in the brodcast address and port on witch the packet will go out to
+  Udp.write( WOL_packet, 102 );                                 // sends out the packet
```
Usage of "magic constant" discouraged.

Astro-Johnny replied:
If you're not afraid of macros `#define ARRAY_SIZE(_a) (sizeof(_a)/sizeof(_a[0]))` is a way to determine length of an array that can be assigned to a constant or used elsewhere.

Astro-Johnny reviewed (commented):
Not bad, but there is some room for improvements.

tomslw committed:
Added modem support

Changed the location of the functions in general
Added all the needed functions for the modem
Placed the WOL packet creation in setup, thanks for that suggestion
Added finding phrases in the SMS message

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,159 @@
+
+EthernetUDP Udp;                                                // udp stuff
+
+int8_t answer;                                                  // its a temp variable to get the response from the modem
```
No need for this to be global. Functions `send_msg()` and `loop()` can have their own local variables, and the `sendATcommand()` already has.

tomslw committed:
Relay added + Readability

Made some Readability changes.
Added Relay to the mix, for force shut downs and an nother way to turn on the PC.
The relay is ment to be connected to the power button of the PC.

tomslw committed:
Bugfix

The pc wold be turned on by the relay and shot with an WOL if you sent an BACKUP_PC_ON via SMS. I see that as a bug.

tomslw committed:
Watchdog added.

I had a problem with the modem needing lots of power and my power supply couldn't handle the spike of currants and the modem just turned off. But not all ways it turned off and just because this device is meant to be always turned on I added a watchdog.

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+
+void send_msg( char sms_text[] ) {
+
+  while ( ( sendATcommand( "AT+CREG?", "+CREG: 0,1", 500 ) || sendATcommand( "AT+CREG?", "+CREG: 0,5", 500 ) ) == 0);
```
You're issuing the `AT+CREG?` here twice, while it could be done once and then it's output compared against both allowed values. Unfortunately `sendATcommand()` doesn't allow you to do this.

I also recommend commenting the purpose of this construct.

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+EthernetUDP Udp;                                                // udp stuff
+
+int8_t answer;                                                  // its a temp variable to get the response from the modem
+int x;                                                          // its basicly used left and right
```
No need for this to be global. It isn't used "left and right", it's only used in `loop()` and should be made local to that function.

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+  memset( WOL_packet, 0, 102 );                                 // sets all WOL_packets values to 0
+  
+  for ( int i = 0; i < 6; i++ ) {                               // sets the first six values of the packet to 0xFF
+    WOL_packet[1] = 0xFF;
```
I wonder how did the WOL packet ever work? :thinking:

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+      digitalWrite( 8, LOW );
+      wdt_enable(WDTO_8S);
+    }
+    while ( 0 == sendATcommand( "AT+CMGD=1,4", "OK", 2000 ) );
```
"Yoda conditional" discouraged. Of course, this is a matter of style, but you don't seem to use "Yoda conditionals" anywhere else in the code.

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+  answer = sendATcommand( "AT+CMGR=1", "+CMGR:", 2000 );
+  if ( answer == 1 ) {
+    answer = 0;
+    while ( Serial.available() == 0 );
```
Redundant construct: this `while` loop disallows entering the following `do` loop if no symbols are buffered on serial line, but it is inconsequential as the `do` loop has an equivalent check of it's own.

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+        SMS[x] = Serial.read();
+        x++;
+        if ( strstr( SMS, "OK" ) != NULL ) {
+          answer = 1;
```
A `break` statement could be used here to exit loop instead.

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+
+int8_t answer;                                                  // its a temp variable to get the response from the modem
+int x;                                                          // its basicly used left and right
+int onModulePin = 2;                                            // pins and stuff
```
`onModulePin` can be `const`

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+    } while ( answer == 0 );
+
+    if ( strstr( SMS, "BACKUP_PC_ON" ) != NULL ) {
+      digitalWrite( 8, HIGH );
```
Avoid "magic constant". Instead of `8` I recommend using a constant variable (much alike already used `onModulePin`). Using a constant var will allow you to change the pin used in a single place (instead changing it everywhere you've written `8`).

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+
+#include <EthernetUdp.h>
+#include <SPI.h>
```
Inclusion of SPI library redundant: no SPI routines used in code.

Astro-Johnny commented on drill/drill.ino:
```diff
@@ -0,0 +1,172 @@
+
+#include <EthernetUdp.h>
```
Inclusion of `EthernetUdp.h` possibly redundant: `EthernetUDP` class moved to `Ethernet.h` since version 2.0.0 (July 2018).

Astro-Johnny reviewed (commented):
Good! Keep working!
